### ISSUE TITLE
high flow breaks down after adding queue-blocktime

### ISSUE THREAD

**Issue by namdre** [eclipse/sumo#7195]
in tests/sumo/basic/dynamics/insertion/high_flow2 the average speed has dropped from 13m/s to 4m/s
after the changes of #7143

The main difference happens at the transition from edge 'beg2'  (a single queue) to 'merge' (multiple queues). 
Vehicles enter segments of 'beg2' in rapid succession due to tau-scaling but cannot leave with the same spacing. This causes the segment to switch into the jammed-state and flow-breakdown commences.

**Comment by namdre** [comment]
problematic lines:
MESegment::send `  lc->setEventTime(MAX2(lc->getEventTime(), q.getBlockTime()));`
The new leader car could use a different queue from the car that just left. This limits flow to 1 lane onto a mult-queue segment

MESegment::hasSpaceFor: ` if (q.allows(svc) && q.size() < minSize) {` is dubious. A queue with equal (or higher?) size could have a lower block time

**Comment by namdre** [comment]
another dubious piece of code is in MESegment::send:
```
         myLastHeadway = next->getTimeHeadway(nextQIdx, this, q.getOccupancy() <= myJamThreshold, veh);                                                                               
         q.setBlockTime(q.getBlockTime() + myLastHeadway);
```
myLastHeadwayTime returns the headway for a subsequent queue but uses this to set the block time on the current queue. When the current segment is single-queue (n lanes) and the next is multi-queue (n lanes) then the capacity of the first segment is reduced to that of a single lane

**Comment by behrisch** [comment]
> problematic lines:
> MESegment::send ` lc->setEventTime(MAX2(lc->getEventTime(), q.getBlockTime()));`
> The new leader car could use a different queue from the car that just left. This limits flow to 1 lane onto a mult-queue segment

No, removeCar makes sure that it is the same queue

> 
> MESegment::hasSpaceFor: ` if (q.allows(svc) && q.size() < minSize) {` is dubious. A queue with equal (or higher?) size could have a lower block time

The check for the entryBlocktime is in MESegment.cpp, line 315 `if (entryTime >= q.getEntryBlockTime())`. I think not the minimum block time shoul decide only whether it is currently blocked.

**Comment by behrisch** [comment]
> 
> 
> another dubious piece of code is in MESegment::send:
> 
> ```
>          myLastHeadway = next->getTimeHeadway(nextQIdx, this, q.getOccupancy() <= myJamThreshold, veh);                                                                               
>          q.setBlockTime(q.getBlockTime() + myLastHeadway);
> ```
> 
> myLastHeadwayTime returns the headway for a subsequent queue but uses this to set the block time on the current queue. When the current segment is single-queue (n lanes) and the next is multi-queue (n lanes) then the capacity of the first segment is reduced to that of a single lane

This has been fixed via 76349bc
